redis/src/main/kotlin/rys/ajaxpetproject/redis/repository/CacheChatRepository.kt

```diff
@@ -0,0 +1,184 @@
+                actualRepository.findChatById(id)
+                    .flatMap { savedChat ->
+                        redisOperations.opsForValue().set("$CHAT_CACHE_KEY_PREFIX$id", savedChat)
+                            .thenReturn(savedChat)
+                            .doOnSuccess {
+                                logger.info("Chat with id {} was saved in cache", savedChat.id)
+                            }
+                    }
+            }
+    }
+
+    override fun save(chat: MongoChat): Mono<MongoChat> {
+        return actualRepository.save(chat)
+            .flatMap { savedChat ->
+                redisOperations.opsForValue().set("$CHAT_CACHE_KEY_PREFIX${savedChat.id!!}", savedChat)
+                    .thenReturn(savedChat)
+            }.doOnSuccess { savedChat ->
+                logger.info("Chat with id {} was saved in cache", savedChat.id)
+            }
+    }
+
+    override fun deleteAll(): Mono<Unit> {
+        return actualRepository.deleteAll()
+            .thenMany(
+                redisOperations.scan(ScanOptions.scanOptions().match("$CHAT_CACHE_KEY_PREFIX*").build())
+            )
+            .flatMap {
+                redisOperations.opsForValue().delete(it)
+            }
+            .then(Unit.toMono())
+            .doOnSuccess { logger.info("All chats were deleted from cache") }
+    }
+
+    override fun update(id: String, chat: MongoChat): Mono<MongoChat> {
+        return actualRepository.update(id, chat)
+            .flatMap {
+                redisOperations.opsForValue()
+                    .set("$CHAT_CACHE_KEY_PREFIX$id", it).then(it.toMono())
+            }
+            .doOnSuccess { logger.info("Chat with id {} was updated in cache", id) }
+    }
+
+    override fun addUser(userId: String, chatId: String): Mono<Unit> {
+        return actualRepository.addUser(userId, chatId)
+            .then(
+                actualRepository.findChatById(chatId).flatMap { updatedChat ->
+                    redisOperations.opsForValue().set("$CHAT_CACHE_KEY_PREFIX${updatedChat.id!!}", updatedChat)
+                        .thenReturn(Unit)
+                }.doOnSuccess { logger.info("User with id {} was added to chat with id {} in cache", userId, chatId) }
+            )
+    }
+
+    override fun removeUser(userId: String, chatId: String): Mono<Unit> {
+        return actualRepository.removeUser(userId, chatId)
+            .then(
+                actualRepository.findChatById(chatId)
+                    .flatMap { updatedChat ->
+                        redisOperations.opsForValue().set("$CHAT_CACHE_KEY_PREFIX${updatedChat.id!!}", updatedChat)
+                            .thenReturn(Unit)
+                    }.doOnSuccess {
+                        logger.info(
+                            "User with id {} was removed from chat with id {} in cache",
+                            userId,
+                            chatId
+                        )
+                    }
```

Comment:
refactor it to make function shorter. applicable to all similar places

Reply:
Done